hostplumber/controllers/hostconfig_controller.go

```diff
@@ -0,0 +1,486 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	plumberv1 "hostplumber/api/v1"
+	hoststate "hostplumber/pkg/hoststate"
+)
+
+const (
+	sysClassNet   = "/host/sys/class/net/"
+	sysPciDrivers = "/host/sys/bus/pci/drivers/"
+	sysPciDevices = "/host/sys/bus/pci/devices/"
+)
+
+// HostConfigReconciler reconciles a HostConfig object
+type HostConfigReconciler struct {
+	client.Client
+	Log         logr.Logger
+	Scheme      *runtime.Scheme
+	NodeName    string
+	currentSpec *plumberv1.HostConfigSpec
+}
+
+// +kubebuilder:rbac:groups=plumber.k8s.pf9.io,resources=hostconfigs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=plumber.k8s.pf9.io,resources=hostconfigs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=*,resources=*,verbs=*
+
+func (r *HostConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("hostconfig", req.NamespacedName)
+
+	var specApplied bool = false
+	var hostConfigReq = plumberv1.HostConfig{}
+	if err := r.Get(ctx, req.NamespacedName, &hostConfigReq); err != nil {
+		log.Error(err, "unable to fetch HostConfig")
+		fmt.Println("Unable to fetch new HostConfig CRD!!!")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	r.currentSpec = &hostConfigReq.Spec
+
+	sriovConfigList := r.currentSpec.SriovConfig
+
+	for _, sriovConfig := range sriovConfigList {
+		var pfName string
+		var pfList []string
+		var err error
+		if sriovConfig.PfName != nil {
+			fmt.Printf("Configuring via PF: %s\n", *sriovConfig.PfName)
```

Comment:
Same as in the other controller, remove the debug statements or use the reconciler logger.

Reply:
@erwinvaneyk These functions are not of type Reconciler struct. What is the way to use logging then?

1. Do I pass around the logr.Logger object as a single parameter to each function (and call chains)?
2. Pass around the Reconciler object as an argument to every function?
2. Should I initialize a global logger? I'm not sure why it wasn't done this way. In Python, i'm used to something like importing a logger, and initializing a global LOG object:

```
from oslo_log import log as logging
...
LOG = logging.getLogger(__name__)
...
def foo():
    LOG.info("Hello world")
```